### PR TITLE
Ignore vertex shader builtins by name

### LINKED ISSUE #18: Examples fail with "ValueError: Unbound vertex attribute "gl_VertexID" at location 0"
Running on a Macbook Pro 2019, Python 3.9, macOS 10.15.7, the examples (tried a few) fail with

```
$ python julia_fractal.py
2022-06-02 21:03:34.519 Python[79599:798132] ApplePersistenceIgnoreState: Existing state will not be touched. New state will be written to (null)
Traceback (most recent call last):
  File "/Users/akx/build/zengl/examples/julia_fractal.py", line 14, in <module>
    scene = ctx.pipeline(
  File "/Users/akx/build/zengl/_zengl.py", line 382, in validate
    raise ValueError(f'Unbound vertex attribute "{name}" at location {location}')
ValueError: Unbound vertex attribute "gl_VertexID" at location 0
```
(please ignore the f string, I'm running #17 right now)

Is `gl_VertexID` supposed to be somehow implicitly bound by... something..?

Commenting out the check makes things work, but it sounds like there's something weird here :)

### PR DESCRIPTION
**Pull request by szabolcsdombi**
This should resolve #18

### CODE REVIEW

File: _zengl.py [hunk 1]
```diff
@@ -115,6 +115,14 @@
     'instance': 1,
 }
 
+VERTEX_SAHDER_BUILTINS = {
```
Comment (akx) [inline _zengl.py:118]:
```suggestion
VERTEX_SHADER_BUILTINS = {
```

File: _zengl.py [hunk 2]
```diff
@@ -333,6 +341,7 @@ def validate(attributes, uniforms, uniform_buffers, vertex_buffers, layout, reso
             'location': obj['location'] + i if obj['location'] >= 0 else -1,
         }
         for obj in attributes for i in range(obj['size'])
+        if obj['name'] not in VERTEX_SAHDER_BUILTINS
```
Comment (akx) [inline _zengl.py:344]:
```suggestion
        if obj['name'] not in VERTEX_SHADER_BUILTINS
```

**Review by akx — CHANGES_REQUESTED** [review]
Small typo but looks good!

Do you know why this is required, though? Seems weird that it'd work fine elsewhere but fails for me :) ... or maybe that's just a Mac OpenGL driver quirk?

**Comment by szabolcsdombi** [comment]
> Do you know why this is required, though? Seems weird that it'd work fine elsewhere but fails for me :) ... or maybe that's just a Mac OpenGL driver quirk?

I tested on windows, ubuntu and mac mini
most exampes worked fine, even with software render in VMs.

Usually drivers differ a tiny bit, even for the core OpenGL spec.

**Pull request merged**